Formula/tcpreplay.rb

```diff
@@ -14,11 +14,13 @@ class Tcpreplay < Formula
   depends_on "libdnet"
 
   def install
+    ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
     system "./configure", "--disable-debug",
                           "--disable-dependency-tracking",
                           "--disable-silent-rules",
                           "--prefix=#{prefix}",
-                          "--enable-dynamic-link"
+                          "--enable-dynamic-link",
+                          "--with-macosx-sdk=#{MacOS.latest_sdk_version}"
```

Comment:
```suggestion
                          "--with-macosx-sdk=#{MacOS.version}"
```
Our CI nodes won't have a `/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk`, which might be why it's trying to fall back to `10.15` and failing.

Reply:
Oh ok that would make sense.  Lemme fix that.

Reply:
Though, if we could just configure the `[i]sysroot`, it's probably better to try to pass `MacOS.sdk_path_if_needed` to that.

Reply:
do you have some pointers on a sysroot?  trying with MacOS.version right now

Reply:
```
4 % ./configure --prefix=/opt/homebrew/Cellar/tcpreplay/4.3.4 --enable-dynamic-link --with-macosx-sdk=11   
checking whether to enable maintainer-specific portions of Makefiles... yes
checking if malloc debugging is wanted... no
checking build system type... arm-apple-darwin20.5.0
checking host system type... arm-apple-darwin20.5.0
checking target system type... arm-apple-darwin20.5.0
checking what macOS compiler to use... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -m64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
checking for cygwin development environment... no
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... config/install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -m64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
checking whether the C compiler works... no
configure: error: in `/private/tmp/tcpreplay-20210614-55472-19kmhgx/tcpreplay-4.3.4':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

Reply:
I have an idea I'm testing.
I think calling `autogen.sh` might help

Reply:
@carlocab

I dunno what to do here.

I've tried several options and it doesn't like 10.15
```
diff --git a/Formula/tcpreplay.rb b/Formula/tcpreplay.rb
index 2cf24c0714..74bc22267f 100644
--- a/Formula/tcpreplay.rb
+++ b/Formula/tcpreplay.rb
@@ -12,15 +12,25 @@ class Tcpreplay < Formula
   end
 
   depends_on "libdnet"
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
 
   def install
     ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
-    system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
-                          "--disable-silent-rules",
-                          "--prefix=#{prefix}",
-                          "--enable-dynamic-link",
-                          "--with-macosx-sdk=#{MacOS.latest_sdk_version}"
+    args = %W[
+      --disable-debug
+      --disable-dependency-tracking
+      --disable-silent-rules
+      --prefix=#{prefix}
+      --enable-dynamic-link
+    ]
+
+    args << "--with-macosx-sdk=#{MacOS.version}" if MacOS.sdk_path_if_needed
+
+    system "./autogen.sh"
+    system "./configure", *args
+
     system "make", "install"
   end
 
```

Is there any reason the arm nodes don't have 11?  I thought the Apple Silicon hardware only ran 11 or greater?

Reply:
They should have it; not sure why `configure` isn't picking it up. Try
```ruby
sdk = MacOS.sdk_path_if_needed
args << "--with-sysroot=#{sdk}" if sdk
```

Reply:
trying ty

Reply:
ok I have an idea, it works if I set `CFLAGS` with `-isysroot`. testing a change

Reply:
```
diff --git a/Formula/tcpreplay.rb b/Formula/tcpreplay.rb
index 2cf24c0714..1edc0ae3e5 100644
--- a/Formula/tcpreplay.rb
+++ b/Formula/tcpreplay.rb
@@ -12,15 +12,32 @@ class Tcpreplay < Formula
   end
 
   depends_on "libdnet"
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
 
   def install
     ENV["MACOSX_DEPLOYMENT_TARGET"] = MacOS.version
-    system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
-                          "--disable-silent-rules",
-                          "--prefix=#{prefix}",
-                          "--enable-dynamic-link",
-                          "--with-macosx-sdk=#{MacOS.latest_sdk_version}"
+    args = %W[
+      --disable-debug
+      --disable-dependency-tracking
+      --disable-silent-rules
+      --prefix=#{prefix}
+      --enable-dynamic-link
+    ]
+
+    # Apple Silicon does not build if it gets the 10.15 sdk
+    sdk = MacOS.sdk_path_if_needed
+    if sdk
+      ENV.append "CFLAGS", "-isysroot #{sdk}"
+      ENV.append "LDFLAGS", "-isysroot #{sdk}"
+      args << "--with-sysroot=#{sdk}"
+      args << "--with-libpcap=#{sdk}/usr"
+    end
+
+    system "./autogen.sh"
+    system "./configure", *args
+
     system "make", "install"
   end
```

I dunno, that's pretty messy, I'm going to see what I can remove, but that worked. I need to test on intel as well, as that's pretty wild.

Reply:
@carlocab I repushed with that change, it's working for me in intel and m1.
I'm going to see what happens with the CI build